cli/command/trust/sign.go

```diff
@@ -50,12 +57,41 @@ func signImage(cli command.Cli, image, tag string) error {
 			fmt.Fprintf(cli.Out(), "Finished initializing %q\n", notaryRepo.GetGUN().String())
 		default:
 			return trust.NotaryError(repoInfo.Name.Name(), err)
+
 		}
 	}
-
-	// TODO: craft and sign the target using image.addTargetToAllSignableRoles (needs to be exported)
+	ctx := context.Background()
+	requestPrivilege := command.RegistryAuthenticationPrivilegedFunc(cli, repoInfo.Index, "push")
+	target, err := createTarget(notaryRepo, tag)
+	if err != nil && strings.Contains(err.Error(), "No valid trust data for") {
+		image.TrustedPush(ctx, cli, repoInfo, ref, *authConfig, requestPrivilege)
+		return nil
+	}
+	err = image.AddTargetToAllSignableRoles(notaryRepo, &target)
+	if err == nil {
+		err = notaryRepo.Publish()
+	}
+	if err != nil {
+		fmt.Fprintf(cli.Out(), "Failed to sign %q:%s - %s\n", repoInfo.Name.Name(), tag, err.Error())
+		return trust.NotaryError(repoInfo.Name.Name(), err)
+	}
+	fmt.Fprintf(cli.Out(), "Successfully signed %q:%s\n", repoInfo.Name.Name(), tag)
 	return nil
 }
+func createTarget(notaryRepo *client.NotaryRepository, tag string) (client.Target, error) {
+	target := &client.Target{}
+	var err error
+	target.Name = tag
+	target.Hashes, target.Length, err = getManifest(notaryRepo, tag)
+	return *target, err
+}
+func getManifest(notaryRepo *client.NotaryRepository, tag string) (data.Hashes, int64, error) {
```

Comment:
nit: since we're getting information necessary to sign the manifest from Notary rather than the manifest itself, perhaps we should rename this to something like `getSignedManifestHashAndSize`?